src/test/java/seedu/address/logic/parser/CommandParserTestUtil.java

```diff
@@ -76,20 +72,16 @@ public static void assertParseBasicFailure(ParserBasic<? extends Command> parser
     }
 
     /**
-     *
+     * Asserts that the parsing of {@code userInputArgs} by {@code findCommandParser} is
+     * successful and the command created has equal {@code personType} and {@code findPredicateMap}
      */
     public static void assertParseFindSuccess(FindCommandParser parser,
-                                              String userInputArgs, Predicate<Person> expectedPredicate,
+                                              String userInputArgs, FindPredicateMap findPredicateMap,
                                               PersonType personType) {
         try {
             FindCommand command = parser.parse(personType, userInputArgs);
             assertEquals(command.getPersonType(), personType);
-            List<Person> testPersons = TypicalPersons.getTypicalPersons();
-            for (Person currPerson : testPersons) {
-                boolean actualRes = command.getPredicate().test(currPerson);
-                boolean expectedRes = expectedPredicate.test(currPerson);
-                assertEquals(expectedRes, actualRes);
-            }
+            assertEquals(command.getPredicate(), findPredicateMap);
         } catch (ParseException pe) {
             throw new IllegalArgumentException("Invalid userInput.", pe);
         }
```

Comment:
Not sure if this method is necessary, it might make more sense to just use the assertParseComplex* functions above since you are no longer using the truth table/result comparison method.

Reply:
I completely agree! Thank you for pointing that out. I have made the changes according to your suggestion.